packages/jest-worker/src/index.js

```diff
@@ -0,0 +1,193 @@
+    if (!exposedMethods) {
+      // $FlowFixMe: This has to be a dynamic require.
+      const child = require(workerPath);
+
+      exposedMethods = Object.keys(child)
+        .filter(name => this._isValidMethod(name))
+        .filter(name => typeof child[name] === 'function');
+
+      if (typeof child === 'function') {
+        exposedMethods.push('default');
+      }
+    }
+
+    exposedMethods.forEach(name => {
+      if (!this._isValidMethod(name)) {
+        throw new TypeError('Cannot define a method called ' + name);
+      }
+
+      this[name] = this._makeCall.bind(this, name);
+    });
+
+    this._stdout = stdout;
+    this._stderr = stderr;
+    this._ending = false;
+    this._cacheKeys = Object.create(null);
+    this._options = options;
+    this._workers = workers;
+  }
+
+  getStdout(): Readable {
+    return this._stdout;
+  }
+
+  getStderr(): Readable {
+    return this._stderr;
+  }
+
+  end() {
+    if (this._ending) {
+      throw new Error('Farm is ended, no more calls can be done to it');
+    }
+
+    const workers = this._workers;
+
+    // We do not cache the request object here. If so, it would only be only
+    // processed by one of the workers, and we want them all to close.
+    for (let i = 0; i < workers.length; i++) {
+      workers[i].send([CHILD_MESSAGE_END, false], this._empty);
+    }
+
+    this._ending = true;
+  }
+
+  // eslint-disable-next-line no-unclear-flowtypes
+  _makeCall(method: string, ...args: Array<any>): Promise<any> {
+    if (this._ending) {
+      throw new Error('Farm is ended, no more calls can be done to it');
+    }
+
+    return new Promise((resolve, reject) => {
+      const {computeWorkerKey} = this._options;
+      const workers = this._workers;
+      const cacheKeys = this._cacheKeys;
+      const request = [CHILD_MESSAGE_CALL, false, method, args];
+
+      let worker = null;
+      let hash = null;
+
+      if (computeWorkerKey) {
+        hash = computeWorkerKey.apply(this, [method].concat(args));
+        worker = hash == null ? null : cacheKeys[hash];
+      }
+
+      // Do not use a fat arrow since we need the "this" value, which points to
+      // the worker that executed the call.
+      function callback(error, result) {
+        if (hash != null) {
+          cacheKeys[hash] = this;
+        }
+
+        if (error) {
+          reject(error);
+        } else {
+          resolve(result);
+        }
+      }
+
+      // If a worker is pre-selected, use it...
+      if (worker) {
+        worker.send(request, callback);
+        return;
+      }
+
+      // ... otherwise use all workers, so the first one available will pick it.
+      for (let i = 0; i < workers.length; i++) {
+        workers[i].send(request, callback);
+      }
+    });
+  }
+
+  _isValidMethod(name) {
+    return !Farm.prototype.hasOwnProperty(name) && !name.startsWith('_');
+  }
+
+  /* istanbul ignore next */
+  _empty() {
+    // Intentionally empty method.
```

Comment:
what is this for and why is it here?

Reply:
This is a nice way of creating an empty method only once.

Reply:
`const emptyFn = () => {};` instead of the instance method.